### ISSUE TITLE
Label feature.node.kubernetes.io/pci-10de.present=true is missing, but my node has T4 GPU

### ISSUE THREAD

**Issue by digger18** [NVIDIA/gpu-operator#78]
I encountered this error:
https://github.com/NVIDIA/gpu-operator/blob/592c2d10a604e4262aaec42a4c6cb9efb4b90254/pkg/controller/clusterpolicy/object_controls.go#L168

But after checking my node labels, I actually had this label on my GPU node.

I started investigating the source code and I found this line:
https://github.com/NVIDIA/gpu-operator/blob/592c2d10a604e4262aaec42a4c6cb9efb4b90254/pkg/controller/clusterpolicy/object_controls.go#L155
I realized my node has to be labeled with `feature.node.kubernetes.io/pci-10de.present=true` for the operator to work correctly,
So I manually labeled the node with `feature.node.kubernetes.io/pci-10de.present=true` and the operator continued it's work to deploy the runtime, driver, plugin and validation containers as expected.

NFD automatically labeled my node with the following labels:
`feature.node.kubernetes.io/pci-0300_1d0f.present=true,feature.node.kubernetes.io/pci-0302_10de.present=true`
But as you can see, `feature.node.kubernetes.io/pci-10de.present=true` is missing.

I am running Ubuntu 18.04 on AWS g4dn.xlarge node.

**Comment by nvjmayo** [comment]
`feature.node.kubernetes.io/pci-0300_1d0f.present=true` is an Amazon VGA id. We can ignore that. (PCI function 0300 is "VGA")

0300_10de and 0302_10de and others would be the for NVIDIA(10de) VGA(0300) or GPU(0302) cards. 

for example, on my desktop system containing an NVIDIA GeForce GT 710B (gk208), I see 0300 (VGA device) and 0403 (Audio device) listed, but not :
```console
$ lspci -n | grep 10de
65:00.0 0300: 10de:128b (rev a1)
65:00.1 0403: 10de:0e0f (rev a1)
```

I would expect NFD to label mine with `feature.node.kubernetes.io/pci-0300_10de.present=true`. 

Either use an older NFD, or also use gpu-feature-discovery to get the older style labeling (pci-10de). 

To close this bug, an assignee will need to implement a small fix to the logic in `pkg/controller/clusterpolicy/object_controls.go` to check for all the possible values (10de, 0300_10de, and 0302_10de). And after that updating `assets/state-monitoring/0900_daemonset.yaml` and other yaml files under `assets/` should finish the job. Assuming gpu-feature-discovery doesn't also need similar changes and can rely entirely on NFD's behavior.

**Comment by shivamerla** [comment]
@digger18 Can you confirm how NFD was deployed on your system? We would need NFD worker to be configured with following options:  ```--options={"sources":{"pci":{"deviceLabelFields":["vendor"]}}}```. This is how its configured when NFD is deployed as part of our GPU operator chart as below in values.yaml. This will allow NFD to label as ```feature.node.kubernetes.io/pci-10de.present=true``` by just using vendor ID of 10de and skipping class id etc.

```
    options:
      sources:
        pci:
          deviceLabelFields:
          - vendor
```

Documentation with gpu-feature-discovery also points to static YAML to deploy NFD with these options:  https://raw.githubusercontent.com/NVIDIA/gpu-feature-discovery/v0.2.0/deployments/static/nfd.yaml. I will double check if this is documented already or else will add a note. Please let me know if this works for you and can close this issue.

**Comment by digger18** [comment]
I used the built-in NFD bundled with GPU Operator.

**Comment by shivamerla** [comment]
@digger18 can you describe the NFD daemonset and attach here to help us verify with same version and options.

**Comment by shivamerla** [comment]
This is fixed with v1.3.0 of operator. gpu-operator now looks into labels with class id of 0300/0302 included and adds a common label nvidia.com/gpu.present=true which is used as node selector by other component pods.
